torch/testing/_internal/common_methods_invocations.py

```diff
@@ -7307,6 +7331,14 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
            dtypes=floating_and_complex_types(),
            sample_inputs_func=sample_inputs_linalg_slogdet,
            decorators=[skipCUDAIfNoMagma, skipCUDAIfRocm, skipCPUIfNoLapack]),
+    OpInfo('linalg.trace',
+           aten_name='linalg_trace',
+           supports_out=False,
+           op=torch.linalg.trace,
+           dtypes=all_types_and_complex_and(torch.bool, torch.bfloat16, torch.float16),
+           backward_dtypes=floating_and_complex_types_and(torch.bfloat16, torch.float16),
```

Comment:
This line may not be necessary (not 100% sure)

Reply:
`torch.linalg.trace` allows non-square matrix as an input.
How about saying that the behavior is equivalent to `torch.diagonal(A, offset, -2, -1).sum(-1)`?

Reply:
This is an extremely odd thing. The trace of a matrix is only mathematically defined for square matrices. I'm opening a PR in the Array API about this because this does not seem right.

I think this comment was supposed to go under the docs comment (which is an answer to a comment above, that's why you cannot answer directly to it here).

Reply:
Let's wait until https://github.com/data-apis/array-api/issues/359 is resolved for this one.